tests/test_metrics.py

```diff
@@ -0,0 +1,54 @@
+import os.path
+
+import numpy as np
+import torch
+
+from utils.metrics import box_area, box_iou, fitness, plot_mc_curve
+
+
+def test_fitness():
+    x = np.zeros((10, 4))
+    result = fitness(x)
+    assert result.tolist() == np.zeros([10]).tolist()
+
+    x = np.ones((10, 4))
+    result = fitness(x)
+    assert result.tolist() == [1.0] * 10
+
+    x = np.ones((1, 4))
+    result = fitness(x)
+    assert result.tolist() == [1.0]
+
+
+def test_box_area():
+    box = np.zeros((4, 10))
+    box[0:2][:] = 0
+    box[2:][:] = 10
+    assert box_area(box).tolist() == [100.0] * 10
+
+
+def test_box_iou():
+    box1 = torch.randn(10, 4)
+    box2 = torch.randn(15, 4)
+
+    box1[0] = torch.as_tensor([0, 0, 10, 10])
+    box2[0] = torch.as_tensor([10, 10, 20, 20])
+
+    box1[1] = torch.as_tensor([0, 0, 10, 10])
+    box2[1] = torch.as_tensor([0, 0, 10, 10])
+
+    box1[2] = torch.as_tensor([0, 0, 10, 10])
+    box2[2] = torch.as_tensor([5, 5, 10, 10])
+
+    iou = box_iou(box1, box2)
+    assert iou.shape == (10, 15)
+    assert iou[0][0] == 0.0
+    assert iou[1][1] == 1.0
+    assert iou[2][2] == 0.25
+
+
+def test_plot_mc_curve(tmpdir):
+    save_dir = tmpdir / "mc_curve.png"
+    px, py = np.random.randn(100), np.random.randn(2, 100)
+    plot_mc_curve(px, py, save_dir=save_dir)
+    assert os.path.exists(save_dir)
```

Comment:
```suggestion
    assert os.path.isdir(save_dir)
```

Reply:
although it is named `dir` but `save_dir` is a path so `isdir` will return `False`